src/client/components/pages/editor-revision.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2019 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import React from 'react';
+import RevisionsTable from './parts/revisions-table';
+import request from 'superagent-bluebird-promise';
+
+
+const {Pager, Button, ButtonGroup, DropdownButton, MenuItem} = bootstrap;
+
+class EditorRevisionPage extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			from: this.props.from,
+			nextEnabled: this.props.results.length === this.props.size,
+			results: this.props.results,
+			size: this.props.size
+		};
+		// React does not autobind non-React class methods
+		this.triggerSearch = this.triggerSearch.bind(this);
+		this.handleClickPrevious = this.handleClickPrevious.bind(this);
+		this.handleClickNext = this.handleClickNext.bind(this);
+		this.handleResultsPerPageChange = this.handleResultsPerPageChange.bind(this);
+	}
+
+	triggerSearch() {
+		const pagination = `size=${this.state.size}&from=${this.state.from}`;
+		request.get(`./revisions/revisions?${pagination}`)
+			.then((res) => JSON.parse(res.text))
+			.then((data) => {
+				this.setState(prevState => ({
+					nextEnabled: data.length >= prevState.size,
+					results: data
+				}));
+			});
+	}
+
+	handleClickPrevious() {
+		this.setState(prevState => ({from: Math.max(prevState.from - prevState.size, 0)}), this.triggerSearch);
+	}
+
+	handleClickNext() {
+		this.setState(prevState => ({from: prevState.from + prevState.size}), this.triggerSearch);
+	}
+
+	handleResultsPerPageChange(value) {
+		this.setState({size: parseInt(value, 10)}, this.triggerSearch);
+	}
+
+	render() {
+		return (
+			<div className="container">
+				<div id="EditorRevisionPage">
+
+					<RevisionsTable results={this.state.results}/>
+					{
+						this.state.results && this.state.results.length ?
+							<div>
+								<hr className="thin"/>
+								<Pager>
```

Comment:
Unless I'm mistaken, the Pager element is the same on the various pages we use it.
It would be a good candidate for a refactoring PR.

I took the liberty to create a ticket for you (don't hesitate to unassign yourself if you don't want to do it :) ) : https://tickets.metabrainz.org/browse/BB-406

Reply:
I am doing this in a different PR :)